test/transformer/descriptor/tsLibs/tsLibs.test.ts

```diff
@@ -147,4 +147,59 @@ describe('typescript lib', () => {
         expect(secondPromise).toBe('');
         expect(secondPromise2).toBe(0);
     });
+
+    it('should set a promise resolved for a promise without type', async () => {
+        interface Interface {
+            // @ts-ignore
+            method(): Promise;
+        }
+
+        const properties: Interface = createMock<Interface>();
+        const result: Interface = await properties.method();
+        expect(result).toBeUndefined();
+    });
+
+    it('should set a promise resolved for a promise with multiple type parameter', async () => {
+        interface Interface<T, T2> {
+            a: T2;
+            method(): Promise<Interface<T, T2>>;
+        }
+
+        const properties: Interface<string, number> = createMock<Interface<string, number>>();
+        const result: Interface<string, number> = await properties.method();
+        expect(result.a).toBe(0);
+    });
+
+    it('should set different promises resolved for a promise with type parameter', async () => {
+        interface Interface<T> {
+            method(): Promise<T>;
+        }
+
+        createMock<Interface<string>>();
+        const propertiesReuse: Interface<number> = createMock<Interface<number>>();
+        const result: number = await propertiesReuse.method();
+        expect(result).toBe(0);
+    });
+
+    it('should set a promise for multiple type reference', async () => {
+        type B<T1> = Promise<T1>;
+        type A<T> = B<T>;
+
+        interface Interface {
+            method(): A<string>;
+        }
+
+        const properties: Interface = createMock<Interface>();
+        const result: string = await properties.method();
+        expect(result).toBe('');
+    });
+
+
+    it('should set a promise resolved for a type mocked directly', async () => {
+        type S<T> = Promise<T>;
+        const properties: S<string> = createMock<S<string>>();
+        createMock<S<number>>();
```

Comment:
I think this line with just `createMock` is redundant

Reply:
Yes. I wanted to make sure that other mocks do not impact, but we should have separate tests for that